### PR TITLE
Include Closed Date of Issues when Fitering them for Changlog Inclusion

### PR DESCRIPTION
**Pull request by SudharakaP**
This checks if the closed date of the issue is greater than or equal to the given updatedSince date of changelog thus eliminating the possibility of getting wrong issues which were updated after they were closed.

Fixes https://github.com/opencollective/opencollective/issues/3080

### CODE REVIEW

**Comment by Betree** [comment]
Thanks @SudharakaP, that's an interesting solution.

However I like that we include issues that were just updated, for example if we forgot to add a `feature` label on an issue it allows us to fix it by adding the label afterwards. 

To address https://github.com/opencollective/opencollective/issues/3080, I would rather make sure that all issues are properly sorted by `closed_at` descending.

**Comment by SudharakaP** [comment]
@Betree : Hmm.... sorry, I might be confused here. We sort by week currently right? So if we sort by closed_at the weeks will be messed up. Like say for example issue https://github.com/opencollective/opencollective/issues/2180 will appear right after 12th week; with the label 37th week. Also there will be more than one instance of the 37th week for each update you do. Does that make sense? :thinking:

**Comment by Betree** [comment]
@SudharakaP I'm not sure to grasp the full complexity, I haven't looked much at the code yet.

We should continue to group by week, but the date used for this should be the date where the issue was closed. The expectations would be something like:

> **Week 3**
> Issue 89 (closed on 2020-01-14)
> 
> **Week 2**
> Issue 10 (closed on 2020-01-11)
> Issue 13 (closed on 2020-01-07)
> 
> **Week 1**
> Issue 52 (closed on 2020-01-01)

**Comment by SudharakaP** [comment]
@Betree : Ah, I think I get it. This is a tricky one. We have several problems here. There's a place we have hard-coded the year to current year and that doesn't handle updates to previous year records properly. Also I don't think the `lodash.uniqBy` which we use to update the previous records isn't exactly correct in my opinion; rather we should use a `lodash.union`. Also probably due to these reasons we had duplicate records in `2019.json`. I've done those changes and also removed the duplicate records from `2019.json`. I think this should properly update the records.

Please let me know if you have questions or see any issues. :smile:

File: scripts/lib/jsonManager.js [hunk 1]
```diff
@@ -19,7 +20,7 @@ export function saveWeekLogs(year, week, issues) {
   const yearLog = logs(year);
   const weekIssues = yearLog.get(weekPath);
   if (weekIssues) {
-    yearLog.set(weekPath, uniqBy([...weekIssues, ...issues], "id"));
+    yearLog.set(weekPath, unionBy([...issues, ...weekIssues], "id"));
```
Comment (Betree) [inline scripts/lib/jsonManager.js:23]: If you pass only one array, `uniqBy` and `unionBy` seems to have the same behavior. The only difference is that you can pass multiple arrays to `union`, ie `unionBy(issues, weekIssues, "id")`.

As per https://lodash.com/docs/4.17.15:
> Union - Creates an array of unique values, in order, **from all given arrays** using SameValueZero for equality comparisons.
> Uniq - Creates a duplicate-free version **of an array**, using SameValueZero for equality comparisons, in which only the first occurrence of each element is kept. The order of result values is determined by the order they occur in the array.

Unless I'm missing something, I'd say we should stick with `uniqBy`.

Comment (SudharakaP) [reply scripts/lib/jsonManager.js]: @Betree : You are right; I was confused here. I was testing with unionBy(issues, weekIssues, "id") and thought I checked that in somehow. :thinking:

**Comment by Betree** [comment]
Let's try :+1:

**Comment by SudharakaP** [comment]
> Let's try +1

Thanks for the review and merge. :smile:

**Pull request merged**